### PR TITLE
Various lib64c-related cleanups

### PR DESCRIPTION
**Pull request by jrtc27**

### CODE REVIEW

File: share/mk/src.opts.mk [hunk 1]
```diff
@@ -216,7 +216,7 @@ __DEFAULT_NO_OPTIONS = \
     ZONEINFO_LEAPSECONDS_SUPPORT \
 
 __DEFAULT_YES_OPTIONS+=	\
-	COMPAT_CHERIABI \
+	LIB64C \
```
Comment (bsdjhb) [inline share/mk/src.opts.mk:219]: Perhaps re-sort the list now?

File: share/mk/bsd.compat.mk [hunk 2]
```diff
@@ -144,13 +144,13 @@ COMPAT_RISCV_MARCH=	rv64ima
...
-.if ${COMPAT_ARCH:Mriscv*c*} || (defined(HAS_COMPAT) && ${HAS_COMPAT:MCHERI})
+.if ${COMPAT_ARCH:Mriscv*c*} || (defined(HAS_COMPAT) && ${HAS_COMPAT:M64C})
 COMPAT_RISCV_MARCH:=	${COMPAT_RISCV_MARCH}xcheri
 .endif
 .endif
 
 # Common CHERI flags
-.if defined(HAS_COMPAT) && ${HAS_COMPAT:MCHERI}
+.if defined(HAS_COMPAT) && ${HAS_COMPAT:M64C}
 LIB64CCFLAGS+=	-DCOMPAT_CHERI
```
Comment (bsdjhb) [inline share/mk/bsd.compat.mk:154]: Hmm, what about COMPAT_CHERI?

**Review by bsdjhb — COMMENTED** [review]
I almost did some of these when doing libcheri -> lib64c.

File: share/mk/bsd.compat.mk (hunk 2 above)
Comment (jrtc27) [reply share/mk/bsd.compat.mk:154]: It's COMPAT_32/64BIT not COMPAT_(LIB)32/64; unclear what the equivalent is so left it alone

Comment (bsdjhb) [reply share/mk/bsd.compat.mk:154]: Ok, I'm fine with leaving it as.

**Review by bsdjhb — APPROVED** [review]

File: release/Makefile [hunk 3]
```diff
@@ -232,12 +232,12 @@ dvd: packagesystem
 		-DDB_FROM_SRC
 # Copy distfiles
 	mkdir -p ${.TARGET}/usr/freebsd-dist
-	for dist in MANIFEST $$(ls *.txz | grep -v -- '(base|lib32|lib64)-dbg'); \
+	for dist in MANIFEST $$(ls *.txz | grep -v -- '(base|lib32|lib64|lib64c)-dbg'); \
```
Comment (brooksdavis) [inline release/Makefile:235]: `-vE` for consistency while you're here?

**Review by brooksdavis — APPROVED** [review]

File: release/Makefile (hunk 3 above)
Comment (jrtc27) [reply release/Makefile:235]: That’s a change of behaviour and means the code isn’t actually doing what is intended upstream. Not sure why it works then, presumably it just doesn’t matter…

Comment (jrtc27) [reply release/Makefile:235]: My guess is the intent is -dbg distfiles require downloading but in fact they’re inadvertently included on dvd but not disc1/memstick?

Comment (bsdjhb) [reply release/Makefile:235]: I think it makes sense to ship them on dvd FWIW, so the upstream fix might be to just "remove dead code".  I would rather not change it here though as part of this, but keep your change as-is and let upstream figure out what it wants to do with dbg dists on dvd.

Comment (jrtc27) [reply release/Makefile:235]: Yeah. The NO_ROOT code below does pass -E though so there's likely a discrepancy there depending on whether you use makefs as root with a real tree or makefs as non-root with a METALOG...

**Pull request merged**